Update EIP-5920: add clarification to ban opcode in STATIC mode (#9589)
ethereum/EIPs: EIPS/eip-5920.md

A new opcode is introduced: `PAY` (`0xfc`), which:

> - Halt with exceptional failure if the current frame is in `static-mode`.

Reviewer: This diction matches EIP-7069 and so this version of the text should be approved.

- Pops two values from the stack: `addr` then `val`.
- Charges the gas cost detailed below.
- Exceptionally halts if `addr` has any of the high 12 bytes set to a non-zero value (i.e. it does not contain a 20-byte address).

> - Exceptionally halts if `val` is nonzero if the `STATIC` flag of the current call frame is `true`

Reviewer 2: I don't think we should complicate it with wether `val` is used or not.  There are better non-writing ways to "warm" a target address.

```suggestion
- halt with exceptional failure if the current frame is in `static-mode`.
```

Author: Ok, this is also fine by me. I copied the behavior of call, which allows calls if no value is sent (this is of course used in static contexts to call libraries for instance). I also used `STATIC` flag which is the original EIP terminology. But I think `static-mode` is also clear for every EVM developer.

Author: Actually, indeed, also for the warm use case: `BALANCE` is clearly used to read, and `PAY` is clearly used to write. So `PAY` should not be used in a static context.
